### ISSUE TITLE
[question] why `nix::ExprPath` requires an `InputAccessor`

### ISSUE THREAD

**Issue by inclyc** [NixOS/nix#10219]
I just see recently https://github.com/NixOS/nix/pull/9177 adds an "input accessor" in `ExprPath`, does this mean each instances of `nix::ExprPath` may have a different "accessor" namespaces?

And, can we assume that in each valid nix expression / file, `ExprPath`s have **shared** input accessor?

CC @edolstra

**Comment by edolstra** [comment]
Yes, they can be different. Currently most `ExprPath`s will refer to the root filesystem accessor, but in the future they will refer to the flake input they came from.

**Comment by inclyc** [comment]
> but in the future they will refer to the flake input they came from.

So, as per my understanding, each ExprPath will share the same "accessor" produced by a parsing session. However accessors may be different among flake inputs.

And, this difference comes from calling `EvalState::parse` with different `rootFS` arguments, it this correct?

**Comment by edolstra** [comment]
Yes, that's correct.
